### ISSUE TITLE
Distinct Message display format in between sender and receiver.

### ISSUE THREAD

**Issue by newETA** [aspnet/SignalR#3006]
Thank you so much for helping.  So far I am new for SignalR and trying to implement a fully functional real-time chat conversation module. 

So far I am able to make a conversation and working nice as expected.  Except for newly send message distinct between users. 

When I load the existing message for the first time that load as expected.   Which is showing below.
<img width="228" alt="signalr" src="https://user-images.githubusercontent.com/42071761/45990088-03de2780-c044-11e8-8f94-693ff4bd550e.PNG">

But when I try to send a new message it's not formatting well since in both clients having the same CSS class name both will have the same type of display format like below.

<img width="232" alt="secondmessage" src="https://user-images.githubusercontent.com/42071761/45990118-3720b680-c044-11e8-8984-34fb2f6c8656.PNG">

Now When I sent a message to another client that should be displayed like the first image shown above. 

I tried to check the condition because of Class both are displaying in same please suggest me how can be accomplished this. 

Requirement:
Sender should always start from Right in Screen.
The receiver should receive a message in right hand side with the avatar.  

Basically like skype messaging system.

**Comment by anurse** [comment]
How are you adding the messages to the DOM? It sounds like you're not applying the correct CSS classes.

**Comment by newETA** [comment]
Thank you for your reply. However, while parsing conversation if the owner is sending a message I am passing "Own" class in the Chat Item but when it's displayed in the chat box it displays "Own" because of that both sides having "Own" class so I am not able to find how to display "Own" class for sender only.  If you see the first Image when I loading in first box initialization that working correctly. Since I am passing whole "Chat Item" in Razor View it keeps "OWN" class and all DOM will be inserted inside chat box with "Own" class name and both side display same class and that is I am not able to figure it out how not to display "Own" class for receiver.

**Comment by anurse** [comment]
How are you passing this information? I need to see sample code, otherwise I can't really identify the issue. Is this a SignalR-specific problem? It sounds like SignalR is not related to what's causing the issue.

**Comment by newETA** [comment]
Thank you again. 

This screen has  content model which I am passing to a partial view which is _MessageSingle.cshtml. 

I tried to set the owner of the content in the model but both sides will have same the content and that one is I am not able to understand how to set that in client side or something.  
<img width="857" alt="sendmessage" src="https://user-images.githubusercontent.com/42071761/46053924-c5fc0480-c112-11e8-8c50-7a90a5945f3d.PNG">

This screen the view I am using to render the model. 
<img width="949" alt="message" src="https://user-images.githubusercontent.com/42071761/46054062-8550bb00-c113-11e8-9d47-ca7f7b031088.PNG">

and the final client is something like. 

<img width="948" alt="clienside" src="https://user-images.githubusercontent.com/42071761/46054111-d82a7280-c113-11e8-9284-3908549b4d9a.PNG">

 this._getSingleChatRoom(data.user, data.room); this method rendering correctly. 

and the problem is in because rendering the same content.  
Realtime.prototype._onSingleMessageReceive = function (roomId, message)

Hope it helps you to give me a clue to get succeed in my SignalR Practice.

**Comment by anurse** [comment]
I don't see how SignalR could be affecting this. I think the issue lies in how you are rendering the post. Does the HTML rendered on the server have the right content? If not, that would be where you need to investigate. Try debugging the server and client code and see if the HTML is what you expect there.

Also, it's generally not a good idea to render HTML on your server like this. Instead, the SignalR service should be sending the **data** down to the client and you should use some kind of JavaScript templating system to render the content there. Otherwise you're sending a lot of HTML back and forth that the client could have just computed itself.

**Comment by newETA** [comment]
HI @anurse  Thank you so much, I make this happen using in the javascript side may be my problem was while rending all in server side. So Thank you so much for your time.
